test_bot/test_bot.py

```diff
@@ -294,15 +294,16 @@ def test_buggy_engine() -> None:
 
     def engine_path(CONFIG: CONFIG_DICT_TYPE) -> str:
         dir: str = CONFIG["engine"]["dir"]
-        name: str = CONFIG["engine"]["name"]
+        name: str = CONFIG["engine"]["name"].removesuffix(".py")
+        if platform == "win32":
+            name += ".bat"
+        else:
+            st = os.stat(os.path.join(dir, name))
+            os.chmod(os.path.join(dir, name), st.st_mode | stat.S_IEXEC)
         return os.path.join(dir, name)
```

Comment:
The repetition of `os.path.join()` can be removed by putting `path = os.path.join(dir, name)` above `if platform == "win32":`.
```python
        path = os.path.join(dir, name)
        if platform == "win32":
            path += ".bat"
        else:
            st = os.stat(path)
            os.chmod(path, st.st_mode | stat.S_IEXEC)
        return path
```